mitodl/open-discussions
# Retry bulk search index tasks on connection errors

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #3947

#### What's this PR do?
Adds retries to all bulk indexing tasks for connection errors.
Adds a configurable timeout setting for Elasticsearch (default is same as builtin ES timeout for now, 10 secs).

#### How should this be manually tested?
Best tested on RC: run `python manage.py recreate_index --courses` and see if it completes without any errors (right now it always triggers some timeout errors).

You should run the same command locally and make sure it works there too.
